server/service/devices.go

```diff
@@ -262,6 +262,35 @@ func listDeviceHostDeviceMappingEndpoint(ctx context.Context, request interface{
 	return listHostDeviceMappingResponse{HostID: host.ID, DeviceMapping: dms}, nil
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Put Current Device's Host Device Mappings
+////////////////////////////////////////////////////////////////////////////////
+
+type putDeviceHostDeviceMappingRequest struct {
+	Token string `url:"token"`
+	Email string `json:"email"`
+}
+
+func (r *putDeviceHostDeviceMappingRequest) deviceAuthToken() string {
+	return r.Token
+}
+
+func putDeviceHostDeviceMappingEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*putDeviceHostDeviceMappingRequest)
+
+	host, ok := hostctx.FromContext(ctx)
+	if !ok {
+		err := ctxerr.Wrap(ctx, fleet.NewAuthRequiredError("internal error: missing host from request context"))
+		return putHostDeviceMappingResponse{Err: err}, nil
+	}
+
+	dms, err := svc.SetCustomHostDeviceMapping(ctx, host.ID, req.Email)
+	if err != nil {
+		return putHostDeviceMappingResponse{Err: err}, nil
+	}
+	return putHostDeviceMappingResponse{HostID: host.ID, DeviceMapping: dms}, nil
+}
+
```

Comment:
All of this file's changes will have to be removed if we go ahead with the orbit endpoint.